src/client/components/ProductDetails/ProductDetails.component.js

```diff
@@ -1,82 +1,106 @@
-                <b>{Price} DKK</b>
+            <small>({RemainingUnit} units left)</small>
+          </div>
+          <div className="price">
+            <b>{Price} DKK</b>
+          </div>
+          <div className="input-row">
+            <div className="select-item">
+              <select onChange={(e) => setColor(e.target.value)}>
+                <option>Color</option>
+                <option value={productColor}>{productColor}</option>
+              </select>
+              <div className="arrow-right" />
             </div>
-            <div className='input-row'>
-                <div className='select-item'>
-                    <select>
-                        <option>Color</option>
-                        <option>{productColor}</option>
-                    </select>
-                    <div className="arrow-right" />
-                </div>
-                <NumberInput initValue={1} maxAvailable={RemainingUnit} />
-            </div>
-            <div className='input-row'>
-                <div className='select-item'>
-                    <select>
-                        <option>Size</option>
-                        <option>{productSize}</option>
-                    </select>
-                    <div className="arrow-right" />
-                </div>
-                <ButtonComponent title='ADD TO CARD' onClick={onClick} />
+            <NumberInput
+              initValue={1}
+              maxAvailable={RemainingUnit}
+              getQuantity={getQuantity}
+            />
+          </div>
+          <div className="input-row">
+            <div className="select-item">
+              <select onChange={(e) => setSize(e.target.value)}>
+                <option>Size</option>
+                <option value={productSize}>{productSize}</option>
+              </select>
+              <div className="arrow-right" />
             </div>
+            <ButtonComponent title="ADD TO CART" onClick={onClick} />
+          </div>
         </div>
-    </div>
-}
+      </div>
+    ),
+  };
+};
 
 ProductDetails.propTypes = {
-    imgSource: PropTypes.string.isRequired,
-    ProductName: PropTypes.string.isRequired,
-    RemainingUnit: PropTypes.number.isRequired,
-    Price: PropTypes.number.isRequired,
-    productColor: PropTypes.string.isRequired,
-    productSize: PropTypes.string.isRequired,
-    onClick: PropTypes.func.isRequired,
-    isFavorite: PropTypes.bool,
-    imageAlt: PropTypes.string,
-}
+  imgSource: PropTypes.string.isRequired,
+  ProductName: PropTypes.string.isRequired,
+  RemainingUnit: PropTypes.number.isRequired,
+  Price: PropTypes.number.isRequired,
+  productColor: PropTypes.string.isRequired,
+  productSize: PropTypes.string.isRequired,
+  onClick: PropTypes.func.isRequired,
+  isFavorite: PropTypes.bool,
+  imageAlt: PropTypes.string,
+};
 
 ProductDetails.defaultProps = {
-    imageAlt: 'Product Image',
-    isFavorite: true,
-}
+  imageAlt: 'Product Image',
+  isFavorite: true,
+};
```

Comment:
What happens in this file? There is looking too many changes in this component

Reply:
ok, i will revert product detail component changes. but without these changes i cant work on add to cart page . that is my next assignment.
Lets discuss on sunday in class.